testsuite/integration/basic/src/test/java/org/jboss/as/test/integration/web/servlet/preservepath/PreservePathTestCase.java

```diff
@@ -95,33 +107,56 @@ public static WebArchive deployment() {
-
-      HttpClient httpclient = HttpClientBuilder.create().build();
-      HttpGet httpget = new HttpGet(url.toString() + "/test?path="+ URLEncoder.encode(tempDir));
-      HttpResponse response = httpclient.execute(httpget);
-
-      Thread.sleep(100);
-      FileReader fr = new FileReader(new File(tempDir + "/output.txt"));
-      Assert.assertEquals("servletPath: /test", new BufferedReader(fr).readLine());
+      runTestPreservePath(true);
    }
 
    @Test
    public void testDontPreservePath() throws Exception {
-      setPreservePathOnForward("false");
+      runTestPreservePath(false);
+   }
+
+   @Test
+   public void testDefaultPreservePath() throws Exception {
+      runTestPreservePath(null);
+   }
+
+   private void runTestPreservePath(Boolean preservePathOnForward) throws Exception {
+      setPreservePathOnForward(preservePathOnForward);
 
       HttpClient httpclient = HttpClientBuilder.create().build();
       HttpGet httpget = new HttpGet(url.toString() + "/test?path="+ URLEncoder.encode(tempDir));
       HttpResponse response = httpclient.execute(httpget);
 
-      Thread.sleep(100);
-      FileReader fr = new FileReader(new File(tempDir + "/output.txt"));
-      Assert.assertEquals("servletPath: /preserve-path.jsp", new BufferedReader(fr).readLine());
+      long end = System.currentTimeMillis() + TIMEOUT;
+      File file = new File(tempDir + "/output.txt");
+      while (!file.exists() && System.currentTimeMillis() < end) {
+         Thread.sleep(100);
+      }
+      Assert.assertTrue(file + " was not created within the timeout", file.exists());
+
+      String expectedServletPath = (preservePathOnForward != null && preservePathOnForward) ?
+              "servletPath: /test" : "servletPath: /preserve-path.jsp";
+      String expectedRequestURL = (preservePathOnForward != null && preservePathOnForward) ?
+              // Note that in the true case, there is an extra slash before test. This might be removed in the future
+              "requestUrl: " + url + "/test" : "requestUrl: " + url + "preserve-path.jsp";
+
+      try (BufferedReader r = new BufferedReader(new FileReader(file))) {
+         String servletPath = r.readLine();
+         Assert.assertEquals(expectedServletPath, servletPath);
+         String requestUrl = r.readLine();
+         Assert.assertEquals(expectedRequestURL, requestUrl);
+
+      }
    }
 
-   private void setPreservePathOnForward(String s) throws Exception {
-      ModelNode setPreservePathOp = createOpNode("system-property=io.undertow.servlet.dispatch.preserve_path_of_forward", ModelDescriptionConstants.ADD);
-      setPreservePathOp.get("value").set(s);
+   private void setPreservePathOnForward(Boolean preservePathOnForward) throws Exception {
+      String opName = preservePathOnForward == null ? UNDEFINE_ATTRIBUTE_OPERATION : WRITE_ATTRIBUTE_OPERATION;
+      ModelNode setPreservePathOp = createOpNode("subsystem=undertow/servlet-container=default", WRITE_ATTRIBUTE_OPERATION);
```

Comment:
Instead of `WRITE_ATTRIBUTE_OPERATION`, should not be used `opName` here?

Reply:
Good catch, I will fix